modules/cstrike/cstrike/CstrikeNatives.cpp

```diff
@@ -1961,7 +1964,15 @@ static cell AMX_NATIVE_CALL cs_get_user_weapon(AMX *amx, cell *params)
 
 	if (!FNullEnt(pWeapon))
 	{
-		return get_pdata<int>(pWeapon, m_iId);
+		int iId = get_pdata<int>(pWeapon, m_iId);
+
+		cell *cpTemp = MF_GetAmxAddr(amx, params[2]);
+		*cpTemp = get_pdata<int>(pWeapon, m_iClip);
+
+		cpTemp = MF_GetAmxAddr(amx, params[3]);
+		*cpTemp = (iId != CSW_KNIFE) ? get_pdata<int>(pPlayer, m_rgAmmo, get_pdata<int>(pWeapon, m_iPrimaryAmmoType)) : 0;
```

Comment:
Instead checking id, I would check if `m_iPrimaryAmmoType` <= 0, because ammo index starts to 1 and you have -1 when the index is not found from ammo name (typically like the knife which doesn't have ammo name) and that would make the code more generic. What do you think?